core/src/main/scala/cats/free/FreeRigidSelective.scala

```diff
@@ -46,22 +46,23 @@ object FreeRigidSelective {
   }
 
   implicit def selective[F[_]: Functor]: Selective[FreeRigidSelective[F, ?]] = new Selective[FreeRigidSelective[F, ?]] {
+
     override def applicative: Applicative[FreeRigidSelective[F, ?]] = new Applicative[FreeRigidSelective[F, ?]] {
       override def map[A, B](fa: FreeRigidSelective[F, A])(f: A => B): FreeRigidSelective[F, B] =
         fa match {
           case x: Pure[F, A] => Pure(f(x.run))
-          case w: Select.Aux[F, A, B] =>
+          case w: Select.Aux[F, _, A] =>
```

Comment:
> The reason this isn't caught by the compiler is that the pattern-matched type was unchecked.  This unchecked refinement type is still unchecked.

Seeing as you don't need the middle type parameter you can switch to `Select[F, A]`, which isn't refined and therefore makes this checked!

Reply:
Thanks, that didn't occur to me!